src/charm.py

```diff
@@ -88,7 +95,7 @@ def _configure(self):
         # check if configuration file has changed and if so push the
         # new config file to the workload container
         prometheus_config = self._prometheus_config()
-        config_hash = str(hashlib.md5(str(prometheus_config).encode('utf-8')))
+        config_hash = hashlib.md5(str(prometheus_config).encode('utf-8')).hexdigest()
```

Comment:
Thanks. Although it may be more appropriate if a  separate PR was created just for this issue so that it can be merged without waiting for this draft PR and besides the two change sets are unrelated.

Reply:
#49